usrp_drivers/n200/n200_driver.cpp

```diff
@@ -316,375 +315,222 @@ void transmit(zmq::context_t &driver_c, USRP &usrp_d, const DriverOptions &drive
-        memcpy ((void *) samples_metadata_size_message.data (), samples_metadata_str.c_str(),
-                samples_metadata_str.size());
+        auto request = RECV_REQUEST(driver_to_dsp, driver_options.get_dsp_to_driver_identity());
+        SEND_REPLY(driver_to_dsp, driver_options.get_dsp_to_driver_identity(), samples_metadata_str);
 
-        data_socket.send(samples_metadata_size_message);
+        request = RECV_REQUEST(driver_to_brian, driver_options.get_brian_to_driver_identity());
+        SEND_REPLY(driver_to_brian, driver_options.get_brian_to_driver_identity(), samples_metadata_str);
+      };
+
+      std::thread post_sqn_work_t(post_sqn_work);
+      post_sqn_work_t.detach();
+
+    }
+
+   DEBUG_MSG(std::endl << std::endl);
 
-      }();
-    );
 
   }
 }
 
+
 /**
- * @brief      Runs in a seperate thread to act as an interface for the ingress and egress data.
+ * @brief      Runs in a seperate thread to control receiving from the USRPs.
  *
  * @param[in]  driver_c        The driver ZMQ context.
+ * @param[in]  usrp_d          The multi-USRP SuperDARN wrapper object.
+ * @param[in]  driver_options  The driver options parsed from config.
  */
-void control(zmq::context_t &driver_c, const DriverOptions &driver_options) {
-  DEBUG_MSG("Enter control thread");
+void receive(zmq::context_t &driver_c, USRP &usrp_d, const DriverOptions &driver_options) {
+  DEBUG_MSG("Enter receive thread");
 
-  DEBUG_MSG("Creating and connecting to thread socket in control");
+  zmq::socket_t start_trigger(driver_c, ZMQ_PAIR);
+  ERR_CHK_ZMQ(start_trigger.bind("inproc://thread"));
 
-  zmq::socket_t driver_packet_pub_socket(driver_c, ZMQ_PUB);
-  ERR_CHK_ZMQ(driver_packet_pub_socket.bind("inproc://threads"))
+  uhd::stream_args_t stream_args("fc32", "sc16");
+  uhd::stream_cmd_t stream_cmd(uhd::stream_cmd_t::STREAM_MODE_START_CONTINUOUS);
+  auto rx_rate_hz = driver_options.get_rx_rate();
 
-  DEBUG_MSG("Creating and binding control socket");
-/*  zmq::socket_t radarctrl_socket(driver_c, ZMQ_PAIR);
-  ERR_CHK_ZMQ(radarctrl_socket.bind(driver_options.get_radar_control_to_driver_address()))
-  zmq::message_t request;
-*/
-  zmq::socket_t data_socket(driver_c, ZMQ_PAIR);
-  ERR_CHK_ZMQ(data_socket.bind("inproc://data"))
+  auto receive_channels = driver_options.get_receive_channels();
+  stream_args.channels = receive_channels;
 
-  zmq::socket_t ack_socket(driver_c, ZMQ_PAIR);
-  ERR_CHK_ZMQ(ack_socket.bind("inproc://ack"))
+  usrp_d.set_rx_rate(rx_rate_hz,receive_channels);  // ~450us
+  uhd::rx_streamer::sptr rx_stream = usrp_d.get_usrp_rx_stream(stream_args);  // ~44ms
 
-  auto identities = {driver_options.get_driver_to_radctrl_identity(),
-                      driver_options.get_driver_to_dsp_identity(),
-                      driver_options.get_driver_to_brian_identity()};
+  auto usrp_buffer_size = 100 * rx_stream->get_max_num_samps();
+  size_t ringbuffer_size = (size_t(500.0e6)/sizeof(std::complex<float>)/usrp_buffer_size) *
```

Comment:
calculation for ringbuffer_size? make a comment about why you need to do it this way (truncation?)